frontend/gateway/gateway.go

```diff
@@ -760,6 +760,17 @@ func (lbf *llbBridgeForwarder) Solve(ctx context.Context, req *pb.SolveRequest)
 		}
 	}
 
+	if req.Definition != nil {
+		// Rewrite digests in the definition. This ensures the digests are validated
```

Comment:
Why is this in gateway and not in solver loader where we already have ops remarshal logic for source policies?

Reply:
I'll update the logic there. I think the logic also needs to be slightly different. The current one has a kind of conditional rewrite and we'll just need to always remarshal it.